src/components/hosts/EditHostForm.tsx

```diff
@@ -0,0 +1,150 @@
+};
+
+type EditHostFormProps = {
+  host: Host;
+  inventory: Inventory;
+  cluster: Cluster;
+  onCancel: () => void;
+  onSuccess: () => void;
+};
+
+const validationSchema = (initialValues: HostUpdateParams, hosts: Host[] = []) =>
+  Yup.object().shape({
+    hostname: hostnameValidationSchema.concat(
+      uniqueHostnameValidationSchema(initialValues.hostname, hosts),
+    ),
+  });
+
+const EditHostForm: React.FC<EditHostFormProps> = ({
+  host,
+  inventory,
+  cluster,
+  onCancel,
+  onSuccess,
+}) => {
+  const dispatch = useDispatch();
+  const { addAlert } = React.useContext(AlertsContext);
+
+  const { requestedHostname } = host;
+  const { hostname } = inventory;
+
+  const initialValues: HostUpdateParams = {
+    hostId: host.id,
+    hostname: requestedHostname || '',
+  };
+
+  const handleSubmit = async (values: HostUpdateParams) => {
+    if (values.hostname === initialValues.hostname) {
+      // no change to save
+      onSuccess();
+      return;
+    }
+
+    const params: ClusterUpdateParams = {};
+    params.hostsNames = [
+      {
+        id: values.hostId,
+        hostname: values.hostname,
+      },
+    ];
+
+    try {
+      const { data } = await patchCluster(cluster.id, params);
+      dispatch(updateCluster(data));
+    } catch (e) {
+      handleApiError(e, () =>
+        addAlert({ title: 'Failed to update host', message: getErrorMessage(e) }),
+      );
+    }
+
+    onSuccess();
+  };
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      initialStatus={{ error: null }}
+      validationSchema={validationSchema(initialValues, cluster.hosts)}
+      onSubmit={handleSubmit}
+    >
+      {({ handleSubmit, status, setStatus }) => (
+        <Form onSubmit={handleSubmit}>
+          <ModalBoxBody>
+            <GridGap>
+              {status.error && (
+                <Alert
+                  variant={AlertVariant.danger}
+                  title={status.error.title}
+                  actionClose={
+                    <AlertActionCloseButton onClose={() => setStatus({ error: null })} />
+                  }
+                  isInline
+                >
+                  {status.error.message}
+                </Alert>
+              )}
+              <StaticTextField
+                name="discoveredHostname"
+                label="Discovered Hostname"
+                value={hostname || ''}
+              />
+              <InputField
+                label="Requested hostname"
+                name="hostname"
+                helperText="This name will replace the original discovered hostname after installation."
+                isRequired
+                isDisabled={!canHostnameBeChanged(host.status)}
+              />
+            </GridGap>
+          </ModalBoxBody>
+          <ModalBoxFooter>
+            <Button
+              key="submit"
+              type={ButtonType.submit}
+              isDisabled={!!status.error /* TODO: use gray */}
```

Comment:
the button should be disabled when `isSubmitting` or `isInvalid`. not status.error